concent_api/common/decorators.py

```diff
@@ -100,14 +104,14 @@ def wrapper(request, *args, **kwargs):
     return wrapper
 
 
-def require_golem_message(view):
+def require_golem_message(view: Callable) -> Callable:
     """
     Decorator for view accepting golem messages
     Unpacks golem message signed with the key it contains
     proof that the client indeed has the private part of that key
     """
     @wraps(view)
-    def wrapper(request, *args, **kwargs):
+    def wrapper(request: HttpRequest, *args: Any, **kwargs: Any) -> Union[HttpResponse, JsonResponse]:
```

Comment:
I'm not sure in this place, you worked with decorators and wrapers longer than me :)

wrapper returns view, so it should be:
`-> Union[HttpResponse, JsonResponse, Callable]:` 

second: if wrapper inside returns `Union[...]`, shouldn't be returned the same by the  `require_golem_message`?

Reply:
> second: if wrapper inside returns Union[...], shouldn't be returned the same by the require_golem_message?

The `require_golem_message` is the actual decorator, and it returns decorated function, that is why returned type is `Callable`.

> wrapper returns view, so it should be:

It returns result of the call to view function (please check the line before last `return view(...)`) so the returned type is either `HttpResponse` or `JsonResponse`.